app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1692,6 +1694,20 @@ class PluginApi {
     downloadCalendar(title, dates);
   }
 
+  /**
+   * Add a function to be called when Enter is pressed on the search menu. If function returns true,
+   * the user will be navigated to full-page search
+   *
+   * Example usage:
+   * ```
+   * api.addForceToFullSearchOnEnterFn((searchMenu) => { return ... });
+   * ```
+   *
+   */
+  addForceToFullSearchOnEnterFn(fn) {
+    addForceToFullSearchOnEnterFn(fn);
+  }
+
```

Comment:
Maybe we broaden the scope of this extension point. I think that something like `searchKeyboardExtensionFn` where we could do something like 

```js
api.searchKeyboardExtensionFn((event) => { 
   if (e.key === "Escape") {
       console.log("foo")
    }
    super() # or leave out to completely override enter behavior 
});
```

I think this would prove useful in the future and should get you to where you need in the meantime.